### PR TITLE
Build pull requests and master with GitHub Actions

### PR DESCRIPTION
**Pull request by quantum5**
After this PR is merged, on every subsequent PR it should say whether the PR builds successfully or not.

Currently, the following components are built:
* [x] client
* [x] kvmfr kernel module
* [x] linux host
* [x] windows host (cross compile)
* [x] obs plugin

Note that in 9ff1859dc134e9f3196703296f868040c77a4dd6, `ivshmemInit` and `ivshmemFree` were split from `ivshmemOpen` and `ivshmemClose`, but only for Windows. To fix the build for the linux host, I created stubs, but the same split should be done at a later time.

### CODE REVIEW

**Comment by JJRcop** [comment]
This should also install `makensis` and build the host installer program

File: .github/workflows/build.yml
Comment (JJRcop) [inline .github/workflows/build.yml:45]: `Configure Linux host`

File: .github/workflows/build.yml
Comment (JJRcop) [inline .github/workflows/build.yml:50]: `Build Linux host`

File: .github/workflows/build.yml
Comment (JJRcop) [inline .github/workflows/build.yml:64]: `Configure Windows host`

File: .github/workflows/build.yml
Comment (JJRcop) [inline .github/workflows/build.yml:69]: `Build Windows host`

**Review by JJRcop — CHANGES_REQUESTED** [review]
Differentiate between Windows and Linux hosts in action names. Right now they are ambiguous

File: .github/workflows/build.yml
Comment (JJRcop) [inline .github/workflows/build.yml:10]: `Install client dependencies`

File: .github/workflows/build.yml
Comment (JJRcop) [inline .github/workflows/build.yml:42]: `Install Linux host dependencies`

File: .github/workflows/build.yml
Comment (JJRcop) [inline .github/workflows/build.yml:61]: `Install Windows host dependencies`

File: .github/workflows/build.yml
Comment (JJRcop) [inline .github/workflows/build.yml:80]: `Install obs plugin dependencies`

**Review by JJRcop — CHANGES_REQUESTED** [review]
Ambiguous dependencies steps, same name each time

**Comment by gnif** [comment]
Agreed with @JJRcop's suggestions, please make these alterations after which I am happy to merge this

**Comment by quantum5** [comment]
Hmm currently doesn't build. Don't merge until investigated.

**Comment by quantum5** [comment]
Okay it should be fine now. The client failing to build is due to needing to update the submodule after gnif/PureSpice#5.

**Pull request merged**
